Source/WebCore/page/WindowFeatures.h

```diff
@@ -39,24 +39,31 @@ namespace WebCore {
 class FloatRect;
 
 struct WindowFeatures {
+    bool wantsPopup { false };
+    bool hasAdditionalFeatures { false };
+
     std::optional<float> x;
     std::optional<float> y;
     std::optional<float> width;
     std::optional<float> height;
 
-    bool menuBarVisible { true };
-    bool statusBarVisible { true };
-    bool toolBarVisible { true };
-    bool locationBarVisible { true };
-    bool scrollbarsVisible { true };
-    bool resizable { true };
+    std::optional<bool> popup;
+    std::optional<bool> menuBarVisible;
+    std::optional<bool> statusBarVisible;
+    std::optional<bool> toolBarVisible;
+    std::optional<bool> locationBarVisible;
+    std::optional<bool> scrollbarsVisible;
+    std::optional<bool> resizable;
 
-    bool fullscreen { false };
-    bool dialog { false };
-    bool noopener { false };
-    bool noreferrer { false };
+    std::optional<bool> fullscreen;
+    std::optional<bool> dialog;
+    std::optional<bool> noopener { std::nullopt };
```

Comment:
Nit: I don't think we need `{ std::nullopt }` here.

Reply:
I had it without before and the builders complained that no values were set for those two.

Reply:
I remember now, it wasn't working because of the message decoder/encoder, all the other member variables are serialized so they are initialized by the decoder to the right value but `noopener` and `no referrer` aren't, just like `additionalFeatures`.